src/GeoJson/CoordinateReferenceSystem/CoordinateReferenceSystem.php

```diff
@@ -90,15 +78,7 @@ final public static function jsonUnserialize($json)
     /**
      * Factory method for creating a CRS object from properties.
      *
-     * This method must be overridden in a child class.
-     *
-     * @see CoordinateReferenceSystem::jsonUnserialize()
      * @param array|object $properties
-     * @return CoordinateReferenceSystem
-     * @throws \BadMethodCallException
      */
-    protected static function jsonUnserializeFromProperties($properties)
-    {
-        throw new \BadMethodCallException(sprintf('%s must be overridden in a child class', __METHOD__));
-    }
+    abstract protected static function jsonUnserializeFromProperties($properties): CoordinateReferenceSystem;
```

Comment:
```suggestion
    abstract protected static function jsonUnserializeFromProperties($properties): self;
```

I saw you used a `self` return type in other contexts (e.g. `GeoJSON::jsonUnserialize()`), so I assume there's no issue doing so here as well. Made similar suggestions for both child classes.

Reply:
Good catch! Done! 💪